### PR TITLE
Integrate Quantile and Ordinal scaling

### LINKED ISSUE #3579: GridLayer: add support for Quantile and Ordinal scaling
- [ ] Add utility methods with docs unit-tests.
  - [x] Provide new scaling options for quantile and ordinal for GridLayer
      - [x] Support these for color scaling in CPUGridLayer
      - [x] Add this as CPU fallback case in GridLayer
      - [ ] Update docs and add unit tests.

### PR DESCRIPTION
**Pull request by Navneet-Sahota**
Close #3579

#### Change List
- Updated Docs
- Add `domain` property on functions
- Add case for CPU fallback

### CODE REVIEW

**Comment by Navneet-Sahota** [comment]
CPUGridLayer  

#### Scale: Linear
CPU-linear
<br />

#### Scale: Quantize
CPU-quantize
<br />

#### Scale: Quantile
CPU-quantile
<br />

#### Scale: Ordinal
CPU-ordinal

File: modules/aggregation-layers/src/utils/scale-utils.js [hunk 1]
```diff
@@ -110,7 +110,12 @@ export function getQuantileScale(domain, range) {
   while (++i < n) {
     thresholds[i - 1] = threshold(sortedDomain, i / n);
   }
-  return value => quantileScale(thresholds, range, value);
+  function scale(value) {
```
Comment (1chandu) [inline modules/aggregation-layers/src/utils/scale-utils.js]: Can you seperate out this utility method change into a different PR and add unit tests.

File: docs/layers/cpu-grid-layer.md [hunk 2]
```diff
@@ -168,6 +168,11 @@ larger than the elevationUpperPercentile will be hidden.
 Filter cells and re-calculate elevation by `elevationLowerPercentile`. Cells with elevation value
 smaller than the elevationLowerPercentile will be hidden.
 
+##### `colorScaleType` (String, optional)
```
Comment (1chandu) [inline docs/layers/cpu-grid-layer.md]: Should be added to grid-layer.md as well.

File: docs/layers/cpu-grid-layer.md [hunk 3]
```diff
@@ -168,6 +168,11 @@ larger than the elevationUpperPercentile will be hidden.
...
+
+* Default: 'quantize'
+
+Color scale type, default is set to 'quantize'. Supported Values are 'quantize', 'linear', 'quantile' and 'ordinal'.
```
Comment (1chandu) [inline docs/layers/cpu-grid-layer.md]: define the props as, `scaling function used to determine the color of the grid cell, default value is ...`

File: modules/aggregation-layers/src/grid-layer/grid-layer.js [hunk 4]
```diff
@@ -63,6 +64,10 @@ export default class GridLayer extends CompositeLayer {
       // accessor for custom color or elevation calculation is specified
       return false;
     }
+    if (colorScaleType === 'quantile' || colorScaleType === 'ordinal') {
```
Comment (1chandu) [inline modules/aggregation-layers/src/grid-layer/grid-layer.js]: We have section in grid-layer.md that lists all fallback cases, we should update with this case.

**Comment by 1chandu** [comment]
Also add test cases, to `grid-layer-spec.js` to verify new fallback cases.

**Comment by 1chandu** [comment]
For `colorScaleType` prop, you need to add unit tests in `cpu-grid-layer.spec.js`, you can look at existing tests for reference.

**Comment by 1chandu** [comment]
Please remove all references to `EnhancedCPUGridLayer`, it was only for testing. Given you are using golden images from this layer, we should be good. Once removed it is good to land.

**Pull request merged**